### PR TITLE
[Havoc DH] add Sinful Brand uptime tracker

### PR DESCRIPTION
**Pull request by ToppleTheNun**
Add Sinful Brand uptime tracker as it's where the majority of HDH damage comes from in raid.

Change the recommendation of casting Sinful Brand on cooldown as it's actually harmful if you're using double legendary (will add a PR for this check later).

**Sinful Brand DoT Uptime included in Checklist**
image

**Substatistic for Sinful Brand DoT Uptime**
image

**Non-Venthyr Havoc DH**
image

### CODE REVIEW

File: analysis/demonhunterhavoc/src/modules/covenants/SinfulBrandUptime.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          t({
+            id: 'demonhunter.havoc.suggestions.sinfulBrand.uptime',
+            message: `${formatPercentage(actual)}% Sinful Brand uptime`,
+          }),
+        )
+        .recommended(`>${formatPercentage(recommended)}% is recommended`),
+    );
+  }
+
+  subStatistic() {
```
Comment (kfinch) [inline analysis/demonhunterhavoc/src/modules/covenants/SinfulBrandUptime.tsx:65]: Is this getting rendered? `subStatistic` doesn't get placed on the page on its own, it has to be called from somewhere else. If you want a standalone stats box, use `statistic()`

**Review by kfinch — CHANGES_REQUESTED** [review]
Overall looks good, but a screenshot would help to show what this looks like (especially because I think your stats box might not be rendered, see the comment below)

File: analysis/demonhunterhavoc/src/modules/covenants/SinfulBrandUptime.tsx (hunk 1 above)
Comment (ToppleTheNun) [reply analysis/demonhunterhavoc/src/modules/covenants/SinfulBrandUptime.tsx:65]: That's what I get for copy and pasting from the CorruptionUptime implementation. 😄

File: analysis/demonhunterhavoc/src/CONFIG.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
+import { ToppleTheNun } from 'CONTRIBUTORS';
 import Expansion from 'game/Expansion';
 import SPECS from 'game/SPECS';
 
 import CHANGELOG from './CHANGELOG';
 
 export default {
   // The people that have contributed to this spec recently. People don't have to sign up to be long-time maintainers to be included in this list. If someone built a large part of the spec or contributed something recently to that spec, they can be added to the contributors list. If someone goes MIA, they may be removed after major changes or during a new expansion.
-  contributors: [],
+  contributors: [ToppleTheNun],
```
Comment (kfinch) [inline analysis/demonhunterhavoc/src/CONFIG.js:9]: Just so you know, you aren't required to put yourself here. If you do, you'll probably get people asking for more changes. If you're fine with that and intend to add more Havoc DH stuff then certainly you can leave this here.

File: analysis/demonhunterhavoc/src/CONFIG.js [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
   expansion: Expansion.Shadowlands,
   // The WoW client patch this spec was last updated.
-  patchCompatibility: null,
+  patchCompatibility: '9.2.5',
```
Comment (kfinch) [inline analysis/demonhunterhavoc/src/CONFIG.js]: I haven't reviewed the content from this spec recently - are things up to date across the board? If you're confident we can put this through as well, but I don't want it to be automatic.

**Comment by kfinch** [comment]
Apart from the above comments I think this is ready to go in, just double checking

File: analysis/demonhunterhavoc/src/CONFIG.js (hunk 3 above)
Comment (ToppleTheNun) [reply analysis/demonhunterhavoc/src/CONFIG.js]: I'll remove this check for the moment and then do a quick run through after my TS PR to make sure everything's good to go for 9.2.5/9.2.7.

File: analysis/demonhunterhavoc/src/CONFIG.js (hunk 2 above)
Comment (ToppleTheNun) [reply analysis/demonhunterhavoc/src/CONFIG.js:9]: I intend to do some more maintenance of the spec; there's some things that are missing that I'd love to improve.

**Review by kfinch — APPROVED** [review]

**Pull request merged**